dashboard/src/components/AppNew/AppNew.tsx

```diff
@@ -46,26 +48,48 @@ class AppNew extends React.Component<IAppNewProps, IAppNewState> {
     selectedBinding: undefined,
     valuesModified: false,
   };
+
   public componentDidMount() {
-    const { chartID, getBindings, getChartVersion, getChartValues, chartVersion } = this.props;
+    const {
+      chartID,
+      fetchChartVersions,
+      getBindings,
+      getChartVersion,
+      getChartValues,
+      chartVersion,
+    } = this.props;
+    fetchChartVersions(chartID);
     getBindings();
     getChartVersion(chartID, chartVersion);
     getChartValues(chartID, chartVersion);
   }
 
   public componentWillReceiveProps(nextProps: IAppNewProps) {
-    const { version, values } = nextProps.selected;
-    if (version && values && !this.state.valuesModified) {
-      this.setState({ appValues: values });
+    const { selectChartVersionAndGetFiles, chartVersion } = this.props;
+    const { versions } = this.props.selected;
+    if (nextProps.chartVersion !== chartVersion) {
+      const cv = versions.find(v => v.attributes.version === nextProps.chartVersion);
+      if (cv) {
+        selectChartVersionAndGetFiles(cv);
+      } else {
+        throw new Error("could not find chart");
+      }
+    } else {
+      const { version, values } = nextProps.selected;
+      if (version && values && !this.state.valuesModified) {
```

Comment:
We could make this else an `else if` and move this condition up, we can destructure `version` and `values` after line 69. That might be easier to read/follow.